src/py21cmfast/src/IonisationBox.c

```diff
@@ -150,7 +154,7 @@ LOG_SUPER_DEBUG("defined parameters");
         }
     }
 
-    fabs_dtdz = fabs(dtdz(redshift));
+    fabs_dtdz = fabs(dtdz(redshift))/1e15; //reduce to have good precision
```

Comment:
I don't quite understand what this does, but it's a MASSIVE change... you sure it makes sense?

Reply:
this is not related to this bug. But when I debug, I found fabs_dtdz is in order of 1e15 while recombination rate is in order of 1e-15, which is also why we have to use double for variables in recombination. Therefore I just added this factor